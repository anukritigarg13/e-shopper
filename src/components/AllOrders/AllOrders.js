import React from 'react';
import PropTypes from 'prop-types';
import AllOrdersTable from '../AllOrdersTable/AllOrdersTable';
import './AllOrders.css';

const AllOrders = ({ allOrders }) => allOrders.map((order) => (
  <AllOrdersTable key={order.id} order={order} />
));
const allOrdersShape = {
  orderId: PropTypes.number.isRequired,
  date: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  items: PropTypes.number.isRequired,
  time: PropTypes.string.isRequired,

};
AllOrders.propTypes = {
  allOrders: PropTypes.arrayOf(PropTypes.shape(allOrdersShape)).isRequired,
};
export default AllOrders;
