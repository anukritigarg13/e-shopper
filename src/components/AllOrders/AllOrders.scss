.orders-item-heading{
    text-align: left;
    padding: 1rem;
    border-bottom: solid 2px lightgray;
    margin: 2rem 5rem;
    font-size: 2rem;
}