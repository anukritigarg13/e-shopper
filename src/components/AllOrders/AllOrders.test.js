/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import AllOrders from './AllOrders';

describe('AllOrders component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  let mockProps;
  beforeAll(() => {
    mockProps = {
      allOrders: [
        {
          items: {
            Fruits: [{
              id: 1, name: 'MOCK', price: 40, count: 5, category: 'Fruits',
            }],
          },
          date: 1615447258012,
          id: 1,
          totalItems: 1,
          totalCost: 340,
        }, {
          items: {
            Fruits: [{
              id: 1, name: 'MOCK', price: 40, count: 5, category: 'Fruits',
            }],
          },
          date: 1615447258012,
          id: 2,
          totalItems: 1,
          totalCost: 200,
        },
      ],
    };
  });
  it('should render two AllOrdersTable components', () => {
    render(<AllOrders {...mockProps} />);
    const orderTableElements = screen.getAllByTestId('orders-table-test');
    expect(orderTableElements.length).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<AllOrders {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
  it('should display the number of orders to be 2', () => {
    render(<AllOrders {...mockProps} />);
    screen.getByText('All Orders(2 Items)');
  });
});
