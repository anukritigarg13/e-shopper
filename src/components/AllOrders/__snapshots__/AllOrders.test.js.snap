// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrders component should match snapshot 1`] = `
<div>
  <div
    data-testid="all-orders"
  >
    <div
      class="orders-item-heading"
    >
      All Orders
      (
      2
       Items)
    </div>
    <div
      class="all-orders-items-table"
    >
      <div
        class="orders-table"
        data-testid="orders-table-test"
      >
        <table
          class="all-orders-table"
        >
          <thead>
            <tr>
              <th>
                ORDER
              </th>
              <th>
                ITEMS
              </th>
              <th>
                DATE
              </th>
              <th>
                AMOUNT
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                1
              </td>
              <td>
                1
              </td>
              <td>
                Thu Mar 11 2021
              </td>
              <td>
                340
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="order-desc-table"
        >
          <table
            data-testid="order-desc-table"
          >
            <thead>
              <tr>
                <th>
                  ITEM DESCRIPTION
                </th>
                <th>
                  UNIT PRICE
                </th>
                <th>
                  QUANTITY
                </th>
                <th>
                  SUBTOTAL
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="category-row"
                  colspan="4"
                >
                  Fruits
                </td>
              </tr>
              <tr>
                <td>
                  MOCK
                </td>
                <td>
                  40
                </td>
                <td>
                  5
                </td>
                <td>
                  200
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="orders-table"
        data-testid="orders-table-test"
      >
        <table
          class="all-orders-table"
        >
          <thead>
            <tr>
              <th>
                ORDER
              </th>
              <th>
                ITEMS
              </th>
              <th>
                DATE
              </th>
              <th>
                AMOUNT
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                2
              </td>
              <td>
                1
              </td>
              <td>
                Thu Mar 11 2021
              </td>
              <td>
                200
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="order-desc-table"
        >
          <table
            data-testid="order-desc-table"
          >
            <thead>
              <tr>
                <th>
                  ITEM DESCRIPTION
                </th>
                <th>
                  UNIT PRICE
                </th>
                <th>
                  QUANTITY
                </th>
                <th>
                  SUBTOTAL
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="category-row"
                  colspan="4"
                >
                  Fruits
                </td>
              </tr>
              <tr>
                <td>
                  MOCK
                </td>
                <td>
                  40
                </td>
                <td>
                  5
                </td>
                <td>
                  200
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
